pkg/kubectl/cmd/util/diff/diffoptions.go

```diff
@@ -0,0 +1,177 @@
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.EnforceNamespace, &o.FilenameOptions).
+		Flatten().
+		Do()
+	err = r.Err()
+	if err != nil {
+		return err
+	}
+
+	err = r.Visit(func(info *resource.Info, err error) error {
+		if err != nil {
+			return err
+		}
+
+		o.DiffBuffer1, err = runtime.Encode(o.Codec, info.VersionedObject)
+		if err != nil {
+			return err
+		}
+
+		// Verify the object exists in the cluster
+		if err := info.Get(); err != nil {
+			if apierrors.IsNotFound(err) {
+				return err
+			} else {
+				return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+			}
+		}
+
+		o.DiffBuffer2, err = kubectl.GetOriginalConfiguration(info.Mapping, info.Object)
+		if err != nil {
+			return cmdutil.AddSourceToErr(fmt.Sprintf("retrieving current configuration of:\n%v\nfrom server for:", info), info.Source, err)
+		}
+		if o.DiffBuffer2 == nil {
+			return cmdutil.UsageError(cmd, "no last-applied-configuration annotation found on resource: %s, to create the annotation, run the command with --create-annotation", info.Name)
+		}
+
+		//decode and delete the annotations fields
+		targetObj, err := runtime.Decode(unstructured.UnstructuredJSONScheme, o.DiffBuffer2)
+		if err != nil {
+			return nil
+		}
+		delete(targetObj.(runtime.Unstructured).UnstructuredContent()["metadata"].(map[string]interface{}), "annotations")
+
+		o.DiffBuffer2, err = runtime.Encode(o.Codec, targetObj)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (o *DiffOptions) Validate(f cmdutil.Factory, cmd *cobra.Command) error {
+	return nil
+}
+
+func (o *DiffOptions) RunDiffLastApplied(f cmdutil.Factory, cmd *cobra.Command) error {
+	var err error
+	o.DiffBuffer1, err = FormatConvert(o.DiffBuffer1, o.Output)
+	if err != nil {
+		return err
+	}
+	o.DiffBuffer2, err = FormatConvert(o.DiffBuffer2, o.Output)
+	if err != nil {
+		return err
+	}
+	diff := cmdutil.NewDefaultCmdTool("diff", f.DiffEnvs())
+
+	tmpFileSlice := []cmdutil.TempFile{
+		//be careful with the order of DiffBuffer1 and DiffBuffer2
+		{fmt.Sprintf("%s-diff-last-applied-original-", filepath.Base(os.Args[0])), o.ext, bytes.NewReader(o.DiffBuffer2)},
+		{fmt.Sprintf("%s-diff-last-applied-modified-", filepath.Base(os.Args[0])), o.ext, bytes.NewReader(o.DiffBuffer1)},
+	}
+	_, _, err = diff.LaunchTempFile(tmpFileSlice...)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func FormatConvert(input []byte, o string) ([]byte, error) {
+	switch o {
+	case "json":
+		jsonBuffer := &bytes.Buffer{}
+		err := json.Indent(jsonBuffer, []byte(input), "", "  ")
+		if err != nil {
+			return nil, err
+		}
+		return jsonBuffer.Bytes(), nil
+	case "yaml":
+		yamlOutput, err := yaml.JSONToYAML([]byte(input))
+		if err != nil {
+			return nil, err
+		}
+		return yamlOutput, nil
+	}
+	return nil, fmt.Errorf("unexpected -o output mode")
```

Comment:
When I doubt I always follow the style guide.
Following the style guide gives a few advantages:

1. Other people will be more likely to re-use our code since it matches their expectations.
2. New developers who are used to the typical style can contribute without additional context.
3. Go tools like "golint" work without any changes.